### PR TITLE
bug(replays): Fix conditions for showing the Configure Session Replay CTA inside Issue Details

### PR DESCRIPTION
**Pull request by ryan953**
The Configure Session Replay panel was appearing too often, specifically on Issues that are not supported by replay, like Python, iOS and all the rest!

Screen Shot 2022-12-06 at 18 13 26

So we've added a check that the platform is indeed JS

Also, i renamed a bunch of vars to make it easier understand what the check is really doing. And added a test case.

### CODE REVIEW

File: static/app/utils/replays/hooks/useReplayOnboarding.tsx [hunk 1]
```diff
@@ -21,34 +24,34 @@ function getProjectList(selectedProjects: PageFilters['projects'], projects: Pro
...
 
-export function useShouldShowOnboarding() {
+export function useHaveSelectedProjectsSentAnyReplayEvents() {
   const {projects} = useProjects();
   const {selection} = usePageFilters();
 
-  const shouldShowOnboardingPanel = useMemo(() => {
-    const projectList = getProjectList(selection.projects, projects);
-    const hasSentOneReplay = projectList.some(project => project.hasReplays);
-    return !hasSentOneReplay;
```
Comment (ryan953) [inline static/app/utils/replays/hooks/useReplayOnboarding.tsx:31]: Inverted this name & condition, which means changing the tests and the two callsites (one inside this file)

File: static/app/components/events/interfaces/breadcrumbs/breadcrumbs.spec.tsx [hunk 2]
```diff
@@ -6,8 +6,8 @@ import {textWithMarkupMatcher} from 'sentry-test/utils';
 import Breadcrumbs from 'sentry/components/events/interfaces/breadcrumbs';
 import {BreadcrumbLevelType, BreadcrumbType} from 'sentry/types/breadcrumbs';
 import {
+  useHaveSelectedProjectsSentAnyReplayEvents,
```
Comment (billyvg) [inline static/app/components/events/interfaces/breadcrumbs/breadcrumbs.spec.tsx:9]: When looking at the import filename, I would expect that it exports a hook named `useReplayOnboarding` -- though I don't think we have an existing convention for naming so not a blocker

File: static/app/utils/replays/hooks/useReplayOnboarding.tsx [hunk 3]
```diff
@@ -21,34 +24,34 @@ function getProjectList(selectedProjects: PageFilters['projects'], projects: Pro
   return selectedProjects.map(id => projectsByProjectId[id]).filter(Boolean);
 }
 
-export function useShouldShowOnboarding() {
+export function useHaveSelectedProjectsSentAnyReplayEvents() {
```
Comment (billyvg) [inline static/app/utils/replays/hooks/useReplayOnboarding.tsx:27]: It's a bit wordy but describes it well and can't think of anything more succinct.

**Review by billyvg — APPROVED** [review]

File: static/app/utils/replays/hooks/useReplayOnboarding.tsx (hunk 3 above)
Comment (ryan953) [reply static/app/utils/replays/hooks/useReplayOnboarding.tsx:27]: yeah i don't like it. but the old name wasn't doing it for me :(
